src/notification.rs

```diff
@@ -1,218 +1,163 @@
-use once_cell::sync::Lazy;
-use url::Url;
+use std::collections::BTreeMap;
+use std::fmt::Write;
 
-static DEFAULT_MUSTACHE_TEMPLATE: Lazy<mustache::Template> = Lazy::new(|| {
-    mustache::compile_str(
-        "
-{{#singlehost}}
-{{.}} changed
-{{/singlehost}}
-{{^singlehost}}
-{{siteamount}} websites changed
-{{/singlehost}}
-
-{{#sites}}
-- {{.}}
-{{/sites}}
-
-{{#commit}}
-See {{.}}
-{{/commit}}
-",
-    )
-    .unwrap()
-});
-
-#[deprecated = "The notification feature will be removed"]
-#[derive(serde::Serialize)]
-pub struct MustacheData {
-    commit: Option<String>,
-    #[deprecated = "use singlehost"]
-    singledomain: Option<String>,
-    singlehost: Option<String>,
-    siteamount: usize,
-
-    #[deprecated = "use hosts"]
-    domains: Vec<String>,
-    hosts: Vec<String>,
-    sites: Vec<Url>,
-}
+use url::Url;
 
-impl MustacheData {
-    pub fn new(commit: Option<String>, changed_urls: Vec<Url>) -> Self {
-        let mut sites = changed_urls;
-        sites.sort_unstable();
-        sites.dedup();
+fn generate_change_lines(mut changed: Vec<Url>) -> String {
+    changed.sort_unstable();
+    changed.dedup();
 
-        let mut hosts = sites
-            .iter()
-            .filter_map(Url::host_str)
-            .map(std::string::ToString::to_string)
-            .collect::<Vec<_>>();
-        hosts.dedup();
-
-        let singlehost = if let [single] = hosts.as_slice() {
-            Some(single.clone())
-        } else {
-            None
-        };
-
-        Self {
-            commit,
-            singledomain: singlehost.clone(),
-            singlehost,
-            siteamount: sites.len(),
-
-            domains: hosts.clone(),
-            hosts,
-            sites,
+    let mut changed_hosts = BTreeMap::<String, Vec<Url>>::new();
```

Comment:
```suggestion
    type HostString = String;
    ...
    let mut changed_hosts = BTreeMap::<HostString, Vec<Url>>::new();
```
I am not convinced my suggestion is needed but it is the second time it took me > 10s to remember what the String is, in this case. If that keeps happening, we should consider the alias.

Reply:
Yes I agree, as said: not convinced myself. Just didn't want to lose this remark :)  - e.g. if `BTreeMap::<String, Vec<Url>>` is used more often in the whole project, one could argue to have an global alias.